src/wallet/transaction.h

```diff
@@ -315,7 +321,9 @@ class CWalletTx
     template<typename T> T* state() { return std::get_if<T>(&m_state); }
 
     bool isAbandoned() const { return state<TxStateInactive>() && state<TxStateInactive>()->abandoned; }
-    bool isConflicted() const { return state<TxStateConflicted>(); }
+    bool isConflicted() const { return state<TxStateConflicted>() || state<TxStateMempoolConflicted>(); }
+    bool isMempoolConflicted() const { return state<TxStateMempoolConflicted>(); }
+    bool isBlockConflicted() const { return state<TxStateConflicted>(); }
```

Comment:
What do you think of renaming the `TxStateConflicted` struct to `TxStateBlockConflicted`, since that one is used for block conflicts and there is now a difference here between block and mempool conflicts?

`//! State of rejected transaction that conflicts with a confirmed block.
struct TxStateConflicted {
...
};`